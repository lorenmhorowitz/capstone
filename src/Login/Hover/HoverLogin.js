import React, { Component } from "react";
import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
import AppBar from "@material-ui/core/AppBar";
import ToolBar from "@material-ui/core/Toolbar/Toolbar";
import TextField from "@material-ui/core/TextField";
import createMuiTheme from "@material-ui/core/styles/createMuiTheme";
import createPalette from "@material-ui/core/styles/createPalette";
import Typography from "@material-ui/core/Typography/Typography";
import hoverLogo from "../../hoverLogo.png";
import Button from "@material-ui/core/Button";
import "../../css/hover.css";

import request from "request";

const CREATE_ACCOUNT_API =
  "https://us-central1-hdqc-capstone.cloudfunctions.net/login";

const muiTheme = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  palette: createPalette({
    primary: {
      main: "#212121"
    }
  })
});

class HoverLogin extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: ""
    };
  }

  handleEmailChange = data => {
    this.setState({ username: data.target.value });
  };

  handlePasswordChange = data => {
    this.setState({ password: data.target.value });
  };

  handleHoverLogin = () => {
    request.post(
      {
        url: CREATE_ACCOUNT_API,
        body: {
          kind: "HoverAuthentication",
          key: this.state.username,
          value: { password: this.state.password }
        }
      },
      function(error, response, body) {
        if (error) {
          console.log("Error logging in ");
          return;
        }

        if (response.statusCode === 200) {
          this.props.history.replace("/home");
        } else {
          console.log("Invalid response");
        }
      }
    );
  };

  render() {
    return (
      <div className="Hover">
        <MuiThemeProvider theme={muiTheme}>
          <div>
            <AppBar>
              <ToolBar>
                <img src={hoverLogo} style={{ width: 130 }} />
                <div style={{ display: "inline", marginLeft: "auto" }}>
                  <Button color="primary" style={{ color: "white" }}>
                    PRODUCT
                  </Button>
                  <Button color="primary" style={{ color: "white" }}>
                    DEMO
                  </Button>
                  <Button color="primary" style={{ color: "white" }}>
                    PRICING
                  </Button>
                  <Button color="primary" style={{ color: "white" }}>
                    LOG IN
                  </Button>
                  <Button
                    variant="outlined"
                    color="secondary"
                    style={{
                      color: "lightgreen",
                      borderColor: "lightgreen"
                    }}
                  >
                    SIGN UP
                  </Button>
                </div>
              </ToolBar>
            </AppBar>
          </div>
        </MuiThemeProvider>
        <div>
          <Typography
            style={{
              marginTop: 150,
              marginLeft: "auto",
              marginRight: "auto",
              display: "inline-block",
              fontSize: 40
            }}
          >
            Log In
          </Typography>
          <form className={this.props.container} noValidate autoComplete="off">
            <TextField
              id="email-field"
              style={{ marginTop: 60, width: 400 }}
              margin="normal"
              variant="outlined"
              name="email"
              label="email"
              className={this.props.textField}
              onChange={this.handleEmailChange}
            />
          </form>
          <form className={this.props.container} noValidate autoComplete="off">
            <TextField
              id="password-field"
              style={{ marginTop: 10, width: 400 }}
              margin="normal"
              variant="outlined"
              name="password"
              label="password"
              type="password"
              className={this.props.textField}
              onChange={this.handlePasswordChange}
            />
          </form>
          <form>
            <Button
              color="secondary"
              style={{
                marginTop: 50,
                color: "white",
                backgroundColor: "#489e5c",
                width: 400,
                height: 50
              }}
              onClick={this.handleHoverLogin}
            >
              Log in
            </Button>
          </form>
        </div>
      </div>
    );
  }
}

export default HoverLogin;
