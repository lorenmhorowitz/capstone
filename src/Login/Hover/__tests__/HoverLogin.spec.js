import React from "react";
import HoverLogin from "../HoverLogin";
import { shallow } from "enzyme";

describe("Hover login component renders the hover login page correctly", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<HoverLogin />);

    expect(wrapper).toMatchSnapshot();
  });
});
