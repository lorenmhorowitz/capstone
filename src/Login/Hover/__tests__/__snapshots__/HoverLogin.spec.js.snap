// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hover login component renders the hover login page correctly renders correctly 1`] = `
<div
  className="Hover"
>
  <div>
    <header
      className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-8 mui-fixed"
    >
      <div
        className="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
      >
        <img
          src="test-file-stub"
          style={
            Object {
              "width": 130,
            }
          }
        />
        <div
          style={
            Object {
              "display": "inline",
              "marginLeft": "auto",
            }
          }
        >
          <button
            className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-text-43 MuiButton-textPrimary-44 MuiButton-flat-46 MuiButton-flatPrimary-47"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-42"
            >
              PRODUCT
            </span>
            <span
              className="MuiTouchRipple-root-187"
            />
          </button>
          <button
            className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-text-43 MuiButton-textPrimary-44 MuiButton-flat-46 MuiButton-flatPrimary-47"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-42"
            >
              DEMO
            </span>
            <span
              className="MuiTouchRipple-root-187"
            />
          </button>
          <button
            className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-text-43 MuiButton-textPrimary-44 MuiButton-flat-46 MuiButton-flatPrimary-47"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-42"
            >
              PRICING
            </span>
            <span
              className="MuiTouchRipple-root-187"
            />
          </button>
          <button
            className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-text-43 MuiButton-textPrimary-44 MuiButton-flat-46 MuiButton-flatPrimary-47"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-42"
            >
              LOG IN
            </span>
            <span
              className="MuiTouchRipple-root-187"
            />
          </button>
          <button
            className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-outlined-49 MuiButton-outlinedSecondary-51"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "borderColor": "lightgreen",
                "color": "lightgreen",
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-42"
            >
              SIGN UP
            </span>
            <span
              className="MuiTouchRipple-root-187"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
  <div>
    <p
      className="MuiTypography-root-70 MuiTypography-body1-79"
      style={
        Object {
          "display": "inline-block",
          "fontSize": 40,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 150,
        }
      }
    >
      Log In
    </p>
    <form
      autoComplete="off"
      noValidate={true}
    >
      <div
        className="MuiFormControl-root-105 MuiFormControl-marginNormal-106"
        style={
          Object {
            "marginTop": 60,
            "width": 400,
          }
        }
      >
        <label
          className="MuiFormLabel-root-116 MuiInputLabel-root-109 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-outlined-115"
          data-shrink={false}
          htmlFor="email-field"
        >
          email
        </label>
        <div
          className="MuiInputBase-root-136 MuiOutlinedInput-root-123 MuiInputBase-formControl-137"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-130"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiNotchedOutline-legend-154"
              style={
                Object {
                  "width": 0.01,
                }
              }
            />
          </fieldset>
          <input
            aria-invalid={false}
            className="MuiInputBase-input-146 MuiOutlinedInput-input-131"
            disabled={false}
            id="email-field"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </form>
    <form
      autoComplete="off"
      noValidate={true}
    >
      <div
        className="MuiFormControl-root-105 MuiFormControl-marginNormal-106"
        style={
          Object {
            "marginTop": 10,
            "width": 400,
          }
        }
      >
        <label
          className="MuiFormLabel-root-116 MuiInputLabel-root-109 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-outlined-115"
          data-shrink={false}
          htmlFor="password-field"
        >
          password
        </label>
        <div
          className="MuiInputBase-root-136 MuiOutlinedInput-root-123 MuiInputBase-formControl-137"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-130"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiNotchedOutline-legend-154"
              style={
                Object {
                  "width": 0.01,
                }
              }
            />
          </fieldset>
          <input
            aria-invalid={false}
            className="MuiInputBase-input-146 MuiOutlinedInput-input-131 MuiInputBase-inputType-149"
            disabled={false}
            id="password-field"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
          />
        </div>
      </div>
    </form>
    <form>
      <button
        className="MuiButtonBase-root-184 MuiButton-root-158 MuiButton-text-160 MuiButton-textSecondary-162 MuiButton-flat-163 MuiButton-flatSecondary-165"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#489e5c",
            "color": "white",
            "height": 50,
            "marginTop": 50,
            "width": 400,
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-159"
        >
          Log in
        </span>
        <span
          className="MuiTouchRipple-root-194"
        />
      </button>
    </form>
  </div>
</div>
`;
